llvm/lib/Target/AMDGPU/SIPeepholeSDWA.cpp

```diff
@@ -451,13 +497,33 @@ bool SDWASrcOperand::convertToSDWA(MachineInstr &MI, const SIInstrInfo *TII) {
   }
   copyRegOperand(*Src, *getTargetOperand());
   if (!IsPreserveSrc) {
-    SrcSel->setImm(getSrcSel());
+    SdwaSel ExistingSel = static_cast<SdwaSel>(SrcSel->getImm());
```

Comment:
Does the machine verifier enforce that the SDWA values are a valid enum value? If not we should make sure it does

Reply:
It doesn't. I have tried to improve this in this PR #128898.